emu/src/cpu/arm7tdmi.rs

```diff
@@ -903,6 +903,26 @@ impl Arm7tdmi {
 
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
+
+    fn long_branch_link(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let h = op_code.get_bit(11);
+        let offset = op_code.get_bits(0..=10);
+
+        if h {
+            let next_instruction =
+                self.registers.program_counter() as u32 + SIZE_OF_THUMB_INSTRUCTION;
+            let lr = self.registers.register_at(REG_LR);
+            let offset = (offset << 1) as u32;
+            self.registers.set_program_counter(lr + offset);
+            self.registers.set_register_at(REG_LR, next_instruction | 1);
+        } else {
+            let offset = (offset << 12) as u32;
+            let pc = self.registers.program_counter() as u32;
+            self.registers.set_register_at(REG_LR, offset + pc);
+        }
```

Comment:
I think we should use `.wrapping_add`.

Imagine this example, which for semplicity is a 4bits example:

`0b0100 + 0b1110` (`4 - 2`). If we only do `+` we have a panic since it overflows but it's ok to overflow since then we would have `0b0010` which is the correct result

Reply:
Right!!